Example/ViewController.swift

```diff
@@ -17,7 +18,38 @@ class ViewController: UITableViewController {
                 self.dismissProgressIndicator()
             case .failure(let error):
                 print(error.localizedDescription)
-                self.displayErrorAlert(title: "Error", message: "Couldn't retrieve Mini App list, please try again later", dismissController: false)
+                self.displayErrorAlert(title: NSLocalizedString("error_title", comment: ""), message: NSLocalizedString("error_list_message", comment: ""), dismissController: false)
+            }
+        }
+    }
+
+    @IBAction func actionShowMiniAppById() {
+        self.displayTextFieldAlert(title: NSLocalizedString("input_miniapp_title", comment: "")) { (_, textField) in
+            self.dismiss(animated: true) {
+                if let textField = textField, let miniAppID = textField.text, miniAppID.count > 0 {
+                    self.showProgressIndicator {
+                        MiniApp.info(miniAppId: miniAppID) { (result) in
+                            self.dismissProgressIndicator {
+                                switch result {
+                                case .success(let responseData):
+                                    self.currentMiniAppInfo = responseData
+                                    self.performSegue(withIdentifier: "DisplayMiniApp", sender: nil)
+                                case .failure(let error):
+                                    print(error.localizedDescription)
+                                    self.displayErrorAlert(
+                                        title: NSLocalizedString("error_title", comment: ""),
+                                        message: NSLocalizedString("error_single_message", comment: ""),
+                                        dismissController: false)
+                                }
+                            }
+                        }
```

Comment:
Maybe we can move this into a separate function?

Reply:
also done, but I don't find it simplifies reading that much :)

Reply:
It was a suggestion, I felt that the method looked little bigger and possibility that it can be broken down

Reply:
some useful stuff: https://arturbosch.github.io/detekt/complexity.html#longmethod